federator/controllers/amkocluster_controller.go

```diff
@@ -172,6 +171,72 @@ func (r *AMKOClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *AMKOClusterReconciler) FederateGSLBConfig(ctx context.Context, memberClusters []KubeContextDetails) error {
+	// Determine the state that we need to federate across all member clusters
+	var currGCList gslbalphav1.GSLBConfigList
+	err := r.List(ctx, &currGCList, &client.ListOptions{
+		Namespace: AviSystemNS,
+	})
+	if err != nil {
+		return fmt.Errorf("cannot list GSLBConfig list on current cluster in %s namespace: %v",
+			AviSystemNS, err)
+	}
+
+	if len(currGCList.Items) > 1 {
+		return fmt.Errorf("more than one GSLBConfig objects are present in the current cluster")
+	}
+
+	// if no GC objects exist, we need to sync all member clusters to the same state
+	// which would mean, we need to delete GCs on all member clusters (if any)
+	if len(currGCList.Items) == 0 {
+		if err = DeleteObjsOnAllMemberClusters(ctx, memberClusters, AviSystemNS, &gslbalphav1.GSLBConfig{}); err != nil {
+			return fmt.Errorf("error in removing GSLBConfig objects on member clusters: %v", err)
+		}
+		return nil
+	}
+
+	// if a GC object exists in the current cluster, we need to make sure that all the member
+	// clusters have only this GC object in the avi-system namespace.
+	if err = FederateGCObjectOnMemberClusters(ctx, memberClusters, currGCList.Items[0].DeepCopy()); err != nil {
```

Comment:
Can you for a uniform naming of the method? either always call it GC or call it GSLBConfig as the parent method name.

Reply:
Sure, will make the names uniform.